btree/btree.c

```diff
@@ -3780,6 +3793,11 @@ int64_t btree_iter_tick(struct m0_sm_op *smop)
 
 	switch (bop->bo_op.o_sm.sm_state) {
 	case P_INIT:
+		if (oi == NULL) {
```

Comment:
Remove if check. The code makes sure that this state is reached only once at the beginning.

Reply:
if (oi==NULL ) is required, because there are various scenarios where we start traversing again from P_INIT. For example if check fails and height has changed , so here we do not want to allocate memory again for oi.

Reply:
done.